### PR TITLE
Iter5

### PR DESCRIPTION
**Pull request by sashaaro**

### CODE REVIEW

File: cmd/shortener/main.go [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+)
+
+func main() {
+	initConfig()
+
+	urlRepo := infra.NewMemURLRepository()
+	log.Fatal(http.ListenAndServe(internal.Config.ServerAddress, infra.CreateServeMux(urlRepo)))
+}
+
+func initConfig() {
```
Comment (arttet) [inline cmd/shortener/main.go:20]: SUGGESTION: it would be better if it's in `flags.go`

File: internal/infra/http.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package infra
```
Comment (arttet) [inline internal/infra/http.go:1]: CLARIFICATION: please clarify and review your project layout.

File: internal/infra/http.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/sashaaro/url-shortener/internal"
+	"github.com/sashaaro/url-shortener/internal/domain"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+func GenShortURLToken() string {
```
Comment (arttet) [inline internal/infra/http.go:16]: CHANGE REQUEST: this is business logic here.

File: internal/infra/http.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		panic(fmt.Errorf("error while retriving random data: %d %v", n, err.Error()))
+	}
+	return base64.URLEncoding.EncodeToString(buf)[:length]
+}
+
+func CreateServeMux(urlRepo domain.URLRepository) *chi.Mux {
+	r := chi.NewRouter()
+	r.Use(middleware.Logger)
+
+	createShortHandler := func(writer http.ResponseWriter, request *http.Request) {
```
Comment (arttet) [inline internal/infra/http.go:31]: CHANGE REQUEST: it should be a method

**Review by arttet — CHANGES_REQUESTED** [review]
Please add
- https://about.codecov.io/blog/getting-started-with-code-coverage-for-golang/
- https://github.com/golangci/golangci-lint

File: internal/infra/http.go (hunk 2 above)
Comment (sashaaro) [reply internal/infra/http.go]: add handlers & adapters packages.

File: internal/infra/http.go (hunk 3 above)
Comment (sashaaro) [reply internal/infra/http.go]: move to adapters

File: internal/infra/http.go (hunk 4 above)
Comment (sashaaro) [reply internal/infra/http.go]: fixed

File: cmd/shortener/main.go (hunk 1 above)
Comment (sashaaro) [reply cmd/shortener/main.go]: created flags.go

File: internal/adapters/repository.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+var _ domain.URLRepository = &memURLRepository{}
+
+type memURLRepository struct {
+	mx       sync.Mutex
+	urlStore map[domain.HashKey]url.URL
+}
+
+func NewMemURLRepository() domain.URLRepository {
+	return &memURLRepository{
+		urlStore: map[domain.HashKey]url.URL{},
```
Comment (arttet) [inline internal/adapters/repository.go:18]:
```suggestion
		urlStore: make(map[domain.HashKey]url.URL),
```

File: internal/handlers/handlers.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
+package handlers
+
+import (
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/sashaaro/url-shortener/internal"
+	"github.com/sashaaro/url-shortener/internal/adapters"
+	"github.com/sashaaro/url-shortener/internal/domain"
+	"io"
```
Comment (arttet) [inline internal/handlers/handlers.go:9]: Please sort these imports
https://stackoverflow.com/questions/38704769/golang-import-grouping-by-package

File: internal/handlers/handlers.go [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+	"net/http"
+	"net/url"
+)
+
+type HTTPHandlers struct {
```
Comment (arttet) [inline internal/handlers/handlers.go:14]:
```suggestion
type Handler struct {
```

**Review by arttet — APPROVED** [review]

**Pull request merged**